platform-sdk/swirlds-platform-core/src/testFixtures/java/com/swirlds/platform/test/fixtures/addressbook/RandomRosterBuilder.java

```diff
@@ -0,0 +1,311 @@
+    }
+
+    /**
+     * Build a random roster given the provided configuration.
+     */
+    @NonNull
+    public Roster build() {
+        final Roster.Builder builder = Roster.newBuilder();
+
+        if (maximumWeight == null && size > 0) {
+            // We don't want the total weight to overflow a long
+            maximumWeight = Long.MAX_VALUE / size;
+        }
+
+        builder.rosterEntries(IntStream.range(0, size)
+                .mapToObj(index -> {
+                    final NodeId nodeId = getNextNodeId();
+                    final RandomRosterEntryBuilder addressBuilder = RandomRosterEntryBuilder.create(random)
+                            .withNodeId(nodeId)
+                            .withWeight(getNextWeight());
+
+                    generateKeys(nodeId, addressBuilder);
+                    return addressBuilder.build();
+                })
+                .toList());
+
+        return builder.build();
+    }
+
+    /**
+     * Set the size of the roster.
+     *
+     * @return this object
+     */
+    @NonNull
+    public RandomRosterBuilder withSize(final int size) {
+        this.size = size;
+        return this;
+    }
+
+    /**
+     * Set the average weight for an address. If the weight distribution strategy is
+     * {@link WeightDistributionStrategy#BALANCED}, all addresses will have this weight. If the weight distribution
+     * strategy is {@link WeightDistributionStrategy#GAUSSIAN}, this will be the mean weight.
+     *
+     * @return this object
+     */
+    @NonNull
+    public RandomRosterBuilder withAverageWeight(final long averageWeight) {
+        this.averageWeight = averageWeight;
+        return this;
+    }
+
+    /**
+     * Set the standard deviation for the weight for an address. Ignored unless the weight distribution strategy is
+     * {@link WeightDistributionStrategy#GAUSSIAN}.
+     *
+     * @return this object
+     */
+    @NonNull
+    public RandomRosterBuilder withWeightStandardDeviation(final long weightStandardDeviation) {
+        this.weightStandardDeviation = weightStandardDeviation;
+        return this;
+    }
+
+    /**
+     * Set the minimum weight for an address. Overrides the weight generation strategy.
+     *
+     * @return this object
+     */
+    @NonNull
+    public RandomRosterBuilder withMinimumWeight(final long minimumWeight) {
+        this.minimumWeight = minimumWeight;
+        return this;
+    }
+
+    /**
+     * Set the maximum weight for an address. Overrides the weight generation strategy.
+     *
+     * @return this object
+     */
+    @NonNull
+    public RandomRosterBuilder withMaximumWeight(final long maximumWeight) {
+        this.maximumWeight = maximumWeight;
+        return this;
+    }
+
+    /**
+     * Set the strategy used for deciding distribution of weight.
+     *
+     * @return this object
+     */
+    @NonNull
+    public RandomRosterBuilder withWeightDistributionStrategy(
```

Comment:
Should we @NonNull the weightDistributionStrategy parameter?

Reply:
Added @NonNull.